Hdworld/storage with existing s3 (#7602)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/use-with-custom-s3/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Use Amplify Storage with any S3 bucket',
  description: 'You can use the Amplify Storage APIs against your own S3 bucket in your account.',
  platforms: [
    'javascript',
    'react',
    'react-native',
    'angular',
    'vue',
    'nextjs',
    'swift',
    'android',
    'flutter'
  ]
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {

>       meta

Reviewer: pages should pass through the platform
```suggestion
      platform: context.params.platform,
      meta
```

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Use Amplify Storage with any S3 bucket',
  description: 'You can use the Amplify Storage APIs against your own S3 bucket in your account.',
  platforms: [
    'javascript',
    'react',
    'react-native',
    'angular',
    'vue',
    'nextjs',
    'swift',
    'android',
    'flutter'
  ]
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

With Amplify Storage APIs, you can use your own S3 buckets instead of the Amplify-created ones.

<Callout>

> **Important:** To utilize the storage APIs with an S3 bucket outside of Amplify, you must have Amplify Auth configured in your project.

Reviewer 2: ```suggestion
**Important:** To utilize the Storage APIs with an S3 bucket outside of Amplify, you must have valid Auth configuration in your project.
```

Im suggesting the change since technically it is possible for customers to bring their own Auth and this is little misleading.

You can refer to [Amazon S3's Policies and Permissions documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-policy-language-overview.html) for more ways to customize access to the bucket.

Next, use the `addOutput` method from the backend definition object to define a custom s3 bucket by specifying the name and region of the bucket in your **amplify/backend.ts** file.

<Callout>

**Important**

You cannot use both a storage backend configured through Amplify and a custom S3 bucket at the same time. 

If you specify a custom S3 bucket, no sandbox storage resource will be created. The provided custom S3 bucket will be used, even in the sandbox environment.

</Callout>

```javascript

import { defineBackend } from '@aws-amplify/backend';
import { auth } from './auth/resource';
import { data } from './data/resource';

const backend = defineBackend({
  auth,
  data,
});

//highlight-start
backend.addOutput({
  storage: {
    aws_region: "<region>",
    bucket_name: "<bucket-name>"
  },
});
//highlight-end

```
<InlineFilter filters={["javascript", "nextjs", "react", "angular", "vue", "react-native", "android", "swift"]}>

> To ensure the local amplify_outputs.json file is up-to-date, you can run [the npx ampx generate outputs command](/[platform]/reference/cli-commands/#npx-ampx-generate-outputs) or download the latest amplify_output.json from the Amplify console as shown below.

Reviewer 2: ```suggestion
To ensure the local amplify_outputs.json file is up-to-date, you can run [the npx ampx generate outputs command](/[platform]/reference/cli-commands/#npx-ampx-generate-outputs) or download the latest amplify_outputs.json from the Amplify console as shown below.
```
